VideoLocker/src/main/java/org/edx/mobile/module/prefs/UserPrefs.java

```diff
@@ -51,20 +52,21 @@ public boolean isDownloadOverWifiOnly() {
      */
     public File getDownloadFolder() {
         ProfileModel profile = getProfile();
-        
-        File android = new File(Environment.getExternalStorageDirectory(), "Android");
-        File downloadsDir = new File(android, "data");
-        File packDir = new File(downloadsDir, context.getPackageName());
-        File edxDir = new File(packDir, profile.username);
-        edxDir.mkdirs();
+        File videosDir = new File(context.getExternalFilesDir(null).getParent(), AppConstants.Folders.VIDEOS);
```

Comment:
Note that you're doing this multiple times in this commit, with the `getParent()` call always having the potential of throwing a `NullPointerException`, so all the instances will need to be addressed together.